tutorials/preprocessing/plot_55_setting_eeg_reference.py

```diff
@@ -0,0 +1,122 @@
+
+###############################################################################
+# Background
+# ^^^^^^^^^^
+#
+# Typically, one of the first steps in processing EEG recordings is to subtract
+# a *reference signal* from each channel. Conceptually, the reference signal
+# represents environmental or equipment noise that affects all sensors
+# approximately equally, and by subtracting it from the signal at each
+# electrode, what is left will be a less noisy representation of brain activity
+# than the original raw signal.
+#
+# Sometimes the subtracted reference signal is the signal from a physical
+# electrode (typical reference electrode placements are the earlobe or the
+# mastoid processes) or the average from a pair of such electrodes. Other
+# times, the subtracted reference signal is the average of signals at all
+# electrodes. MNE-Python supports all of these possibilities through the
+# :meth:`~mne.io.Raw.set_eeg_reference` method. Of course, you can only set
+# the reference to a specific (set of) electrode(s) if those electrodes are
+# present in the :class:`~mne.io.Raw` file; the :ref:`example data
+# <sample-dataset>` has numeric EEG channel names (``EEG 001``, etc) so the
+# double mastoid and earlobe examples are commented out here:
+
+# use average of mastoid channels as reference
+# raw.set_eeg_reference(ref_channels=['M1', 'M2'])
+
+# use a single channel reference (left earlobe)
+# raw.set_eeg_reference(ref_channels=['A1'])
+
+# use the average of all channels as reference
+raw_avg_ref = raw.copy().set_eeg_reference(ref_channels='average')
+
+###############################################################################
+# When setting ``ref_channels='average'``, MNE-Python will automatically
+# exclude any channels listed in ``raw.info['bads']`` from contributing to the
+# average reference.
+#
+# Creating the average reference as a projector
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# If using an average reference, it is possible to create the reference as a
+# :term:`projector` rather than subtracting the reference from the data
+# immediately, by specifying ``projection=True``:
+
+raw.set_eeg_reference('average', projection=True)
+print(raw.info['projs'])
+
+###############################################################################
+# Creating the average reference as a projector has a couple of advantages:
+#
+# 1. it is possible to turn projectors on or off when plotting, so it is easy
+#    to visualize the effect that the reference has on the data
+#
+# 2. if there are other unapplied projectors affecting the EEG channels (such
+#    as SSP projectors for removing heartbeat or blink artifacts), EEG
+#    referencing cannot be performed until those projectors are either applied
+#    or removed; adding the EEG reference as a projector is not subject to that
+#    constraint. (The reason this wasn't a problem when we created
+#    ``raw_avg_ref`` above is that the projectors )
```

Comment:
The sentence in parentheses is unfinished. Also, I'd start the paragraphs with capital letters.

Reply:
oops. will fix.